dali/operators/reader/coco_reader_op_test.cc

```diff
@@ -112,10 +112,15 @@ class CocoReaderTest : public ::testing::Test {
 
     vector<int> ids(shape.size());
 
-    MemCopy(
-      ids.data(),
-      output.data<int>(),
-      shape.size() * sizeof(int));
+    auto *out_ptr = ids.data();
+    for (int sample_idx = 0; sample_idx < shape.num_samples(); sample_idx++) {
+      MemCopy(
+        out_ptr,
+        output.tensor<int>(sample_idx),
+        shape[sample_idx].num_elements() * sizeof(int));
+      out_ptr += shape[sample_idx].num_elements();
+    }
+    cudaStreamSynchronize(0);
```

Comment:
You don't need that. You are using pagable memory in the copy so it syncs by definition.